dept44-common-validators/src/main/java/se/sundsvall/dept44/common/validators/annotation/ValidMSISDN.java

```diff
@@ -10,23 +8,24 @@
 
 import jakarta.validation.Constraint;
 import jakarta.validation.Payload;
+import se.sundsvall.dept44.common.validators.annotation.impl.ValidMSISDNConstraintValidator;
 
 /**
- * The annotated element must be a valid MSISDN according to the regular expression ^\+[1-9]{1}[0-9]{3,14}$. Accepts
+ * The annotated element must be a valid MSISDN according to the regular expression ^\+[1-9][\d]{3,14}$. Accepts
  * CharSequence.
  */
 @Documented
-@Target({ElementType.FIELD, ElementType.CONSTRUCTOR, ElementType.PARAMETER, ElementType.TYPE_USE})
+@Target({ ElementType.FIELD, ElementType.CONSTRUCTOR, ElementType.PARAMETER, ElementType.TYPE_USE })
 @Retention(RetentionPolicy.RUNTIME)
-@Constraint(validatedBy = ValidMobileNumberConstraintValidator.class)
+@Constraint(validatedBy = ValidMSISDNConstraintValidator.class)
 public @interface ValidMSISDN {
 
 	/**
 	 * Returns the message.
 	 *
 	 * @return the message.
 	 */
-	String message() default "must be a valid MSISDN, regular expression ^\\+[1-9]{1}[0-9]{3,14}$";
+	String message() default "must be a valid MSISDN, regular expression ^\\+[1-9][\\d]{3,14}$";
```

Comment:
Should message this also include some non-technical explanation of what a valid MSISDN could look like?

Reply:
Added an example of a valid MSISDN.